core/templates/dev/head/pages/exploration-editor-page/settings-tab/settings-tab.directive.ts

```diff
@@ -118,6 +118,324 @@ angular.module('oppia').directive('settingsTab', [
+                $window.location = CREATOR_DASHBOARD_PAGE_URL;
+              });
+            }, function() {
+              // Promise is returned by uimodal which is handled here.
+              // No further action is needed.
+            });
+          };
+
+          ctrl.unpublishExplorationAsModerator = function() {
+            AlertsService.clearWarnings();
+
+            var moderatorEmailDraftUrl = '/moderatorhandler/email_draft';
+
+            $http.get(moderatorEmailDraftUrl).then(function(response) {
+              // If the draft email body is empty, email functionality will
+              // not be exposed to the mdoerator.
+              var draftEmailBody = response.data.draft_email_body;
+
+              $uibModal.open({
+                bindToController: {},
+                templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                  '/pages/exploration-editor-page/settings-tab/templates/' +
+                  'moderator-unpublish-exploration-modal.template.html'),
+                backdrop: true,
+                resolve: {
+                  draftEmailBody: function() {
+                    return draftEmailBody;
+                  }
+                },
+                controllerAs: '$ctrl',
+                controller: [
+                  '$uibModalInstance', 'draftEmailBody',
+                  function($uibModalInstance, draftEmailBody) {
+                    var ctrl = this;
+                    ctrl.willEmailBeSent = Boolean(draftEmailBody);
+                    ctrl.emailBody = draftEmailBody;
+
+                    if (ctrl.willEmailBeSent) {
+                      ctrl.EMAIL_BODY_SCHEMA = {
+                        type: 'unicode',
+                        ui_config: {
+                          rows: 20
+                        }
+                      };
+                    }
+
+                    ctrl.reallyTakeAction = function() {
+                      $uibModalInstance.close({
+                        emailBody: ctrl.emailBody
+                      });
+                    };
+
+                    ctrl.cancel = function() {
+                      $uibModalInstance.dismiss('cancel');
+                      AlertsService.clearWarnings();
+                    };
+                  }
+                ]
+              }).result.then(function(result) {
+                ExplorationRightsService.saveModeratorChangeToBackend(
+                  result.emailBody);
+              });
+            });
+          };
+
+          ctrl.isExplorationLockedForEditing = function() {
+            return ChangeListService.isExplorationLockedForEditing();
+          };
+
+          ctrl.closeRolesForm = function() {
+            ctrl.isRolesFormOpen = false;
+          };
+
+          ctrl.isTitlePresent = function() {
+            return ExplorationTitleService.savedMemento.length > 0;
+          };
+
           this.$onInit = function() {
             ctrl.EXPLORATION_TITLE_INPUT_FOCUS_LABEL = (
```

Comment:
You can move the constants here into the top, after "var ctrl = this;", similar to the example in the documentation